sklearn/utils/tests/test_array_api.py

```diff
@@ -164,6 +167,31 @@ def test_asarray_with_order_ignored():
     assert not X_new_np.flags["F_CONTIGUOUS"]
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "sample_weight, normalize, expected",
+    [
+        (None, False, 10.0),
+        (None, True, 2.5),
+        ([0.1, 0.2, 0.3, 0.4], False, 3.0),
+        ([0.1, 0.2, 0.3, 0.4], True, 3.0),
+    ],
```

Comment:
Would it be a good idea to add a case where the sum of sample weights is more than 1? Something like
```suggestion
        ([0.4, 0.4, 0.5, 0.7], True, 2.75),
    ],
```